src/main/java/Command.java

```diff
@@ -0,0 +1,308 @@
+        String toReturn = tasks.deleteTask(tasks, splitInput, ui);
+        storage.updateTaskFile(tasks);
+        return toReturn;
+    }
+
+    /**
+     * Calls for the required methods to handle a done situation.
+     *
+     * @param splitInput A valid string array is of length 2, index 1 being the task number
+     *     to denote as done.
+     * @param tasks The current list of tasks.
+     * @param storage The storage used.
+     * @return The done task message.
+     * @throws InvalidFormatException If the input format is wrong.
+     */
+    public String doneCommand(String[] splitInput, TaskList tasks, Storage storage)
+            throws InvalidFormatException {
+        //checks the formatting of user input
+        if (splitInput.length > 2) {
+            return ui.showErrorMessage(new InvalidFormatException("Please use the correct format:"
+                    + " done <task number>").getMessage());
+        }
+        String toReturn = tasks.doneTask(splitInput, tasks, ui);
+        storage.updateTaskFile(tasks);
+        return toReturn;
+    }
+
+    /**
+     * Calls for the required methods to handle a task adding situation.
+     *
+     * @param splitInput A valid string array is of length 2, index 1 being the task number to
+     *     denote as done.
+     * @param keyWord The type of task.
+     * @param tasks The current list of tasks.
+     * @param storage The storage used.
+     * @return The added task message.
+     * @throws DukeException If any Duke-type exceptions are encountered.
+     */
+    public String addTaskCommand(String[] splitInput, String keyWord, TaskList tasks, Storage storage)
+            throws DukeException {
+        String[] data = processUserTaskInput(splitInput);
+        String toReturn = tasks.addTask(data, keyWord, tasks, ui);
+        storage.updateTaskFile(tasks);
+        return toReturn;
+    }
+
+    /**
+     * Obtains the keyword in the user input to determine command needed.
+     *
+     * @param input The given keyword from user input.
+     * @return The keyword if it is valid.
+     * @throws InvalidKeyWordException If the keyword not valid.
+     */
+    public String getKeyWord(String input) throws InvalidKeyWordException {
+        String[] splitInput = splitUserInput(input);
+        if (parser.isValidKeyWord(splitInput[0])) {
+            return splitInput[0];
+        } else {
+            throw new InvalidKeyWordException("\u2639 OOPS!!! I'm sorry, but I don't know what "
+                    + "that means :-(");
+        }
+    }
+
+    /**
+     * Processes the user input (for task commands only).
+     *
+     * @param array Contains strings for task type, description, date/time if applicable.
+     * @throws InvalidFormatException If the input format is wrong.
+     * @throws EmptyTaskException If input task is not tagged with description.
+     */
+    public static String[] processUserTaskInput(String[] array) throws InvalidFormatException,
```

Comment:
This method seems to be particularly long and quite convoluted. Perhaps you can consider the following methods to make it more readable:
1. Break the code into logic blocks and abstract them into separate methods.
2. Write more comments to explain what each logic block handles. 
3. Use more switch-case statements or else-if statements, instead of many nested if and else statements.

Reply:
I noted the issue before and will resolve if time permits, thanks!